### PR TITLE
CAMEL-14208 camel-undertow: add option to secure endpoints with Keycl…

### PR DESCRIPTION
**Pull request by JiriOndrusek**
Issue: https://issues.apache.org/jira/browse/CAMEL-14208

Added SPI API for plugging security providers into camel-undertow component.

Doc is updated, JUnit test created.
Changes in camel-elytron are not so important. It will be hardly refactored in follow-up fix (see https://issues.apache.org/jira/browse/CAMEL-14743).

### CODE REVIEW

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java [hunk 1]
```diff
@@ -18,11 +18,7 @@
 
 import java.net.URI;
 import java.net.URISyntaxException;
-import java.util.HashMap;
-import java.util.LinkedHashSet;
-import java.util.Locale;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
```
Comment (omarsmak) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: Is the checkstyle enabled? That shouldn't be there

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java [hunk 2]
```diff
@@ -17,8 +17,7 @@
 package org.apache.camel.component.undertow;
 
 import java.net.URI;
-import java.util.Locale;
-import java.util.Map;
+import java.util.*;
```
Comment (omarsmak) [inline components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowEndpoint.java]: here as well

File: components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java (hunk 1 above)
Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: I'll fix it. I've run mvn checkstyle:check before commit without an error...

Comment (omarsmak) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: Is it the same as ` mvn clean install -Psourcecheck`?

Comment (JiriOndrusek) [reply components/camel-undertow/src/main/java/org/apache/camel/component/undertow/UndertowComponent.java]: I hope so, but I'll try it. Both occurrences are fixed.

**Pull request merged**